src/funcionalidades/ban_hammer.py

```diff
@@ -20,14 +32,27 @@ async def on_message(self, message):
         if message.author == self.bot.user:
             return
 
+        message_content = message.content
         forbidden_words_used = [
-            i for i in self.blacklist if message.content.casefold().count(i) > 0]
+            i for i in self.blacklist if message_content.casefold().count(i) > 0]
 
-        if(len(forbidden_words_used) > 0):
-            await message.delete()  # Delete the message
-            # Send a message on the same channel
-            await message.channel.send(message.author.mention + ", debes cuidar tu vocabulario, jovencito")
-            await message.author.send("El mensaje \n" + str(""f"```css\n{message.content}```""") + "no se ajusta a las normas, en los próximos mensajes no uses: " + str(forbidden_words_used).strip('[]'))  # Send a private message to the user
+        if len(forbidden_words_used) > 0:
+            censor_command = self.bot.get_command('censor')
+            censor_command_names = censor_command.aliases
+            censor_command_names.append(censor_command.name)
+            str_ = message_content.split()[0]
+            # If one with permitted roles is baning a word, we dont ban his message
+            if have_permitted_rol(message.author.roles) and str_[0] == self.bot.command_prefix \
+                    and str_[1:] in censor_command_names:
+                pass
```

Comment:
Creo que el pass sobra, yo pondría:
if not(have_permitted_rol(message.author.roles) and str_[0] == self.bot.command_prefix \
                    and str_[1:] in censor_command_names):
 await message.delete()  # Delete the message
                # Send a message on the same channel
                await message.channel.send(message.author.mention + ", debes cuidar tu vocabulario, jovencito")
                # Send a private message to the user
                await message.author.send("El mensaje \n" + str(f'```diff\n-"{message_content}"```') +
                                          "no se ajusta a las normas, en los próximos mensajes no uses: " +
                                          str(forbidden_words_used).strip('[]'))